experiment/dash/dash_otherwise.go

```diff
@@ -0,0 +1,157 @@
+type Simple struct {
+	ConnectLatency  float64 `json:"connect_latency"`
+	MedianBitrate   int64   `json:"cedian_bitrate"`
+	MinPlayoutDelay float64 `json:"min_playout_delay"`
+}
+
+// TestKeys contains the test keys
+type TestKeys struct {
+	Simple Simple `json:"simple"`
+
+	ClientResolver string `json:"client_resolver"`
+
+	// Failure is the failure string
+	Failure string `json:"failure"`
+
+	ReceiverData []neubotModel.ClientResults `json:"receiver_data"`
+}
+
+func measure(
+	ctx context.Context, sess *session.Session, measurement *model.Measurement,
+	callbacks handler.Callbacks, config Config,
+) error {
+	testkeys := &TestKeys{}
+	measurement.TestKeys = testkeys
+	ctx, cancel := context.WithTimeout(context.Background(), defaultTimeout)
+	defer cancel()
+	client := client.New(sess.SoftwareName, sess.SoftwareVersion)
+	// StartDownload starts the DASH download. It returns a channel where
+	// client measurements are posted, or an error.
+	ch, err := client.StartDownload(ctx)
+	if err != nil {
+		testkeys.Failure = err.Error()
+		return err
+	}
+	callbacks.OnProgress(0, fmt.Sprintf("server: %s", client.FQDN))
+	var rates []float64
+	var frameReadyTime float64 = 0.0
+	var playTime float64 = 0.0
+	percentage := 0.0
+	step := 100 / (totalStep + 1) / 100
+	for results := range ch {
+		percentage += step
+		message := fmt.Sprintf(
+			"rate: %s/s speed: %s/s elapsed: %.2f s",
+			humanize.Bytes(uint64(results.Rate*1000/8)), // Rate is kbit in SI size
+			humanize.Bytes(uint64(float64(results.Received)/results.Elapsed)),
+			results.Elapsed,
+		)
+		callbacks.OnProgress(percentage, message)
+		data, err := json.Marshal(results)
+		if err != nil {
+			testkeys.Failure = err.Error()
+			return err
+		}
+		sess.Logger.Debugf("%s", string(data))
+
+		// Here we're computing stats inline
+		// rates is used to calculate MedianBitrate.
+		rates = append(rates, float64(results.Rate))
+
+		if testkeys.Simple.ConnectLatency == 0.0 {
+			// It is always equal for all the records
+			testkeys.Simple.ConnectLatency = results.ConnectTime
+		}
+
+		frameReadyTime += float64(results.Elapsed)
+		if playTime == 0 {
+			playTime += frameReadyTime
+		} else {
+			playTime += float64(results.ElapsedTarget)
+		}
+		var stall float64 = frameReadyTime - playTime
+		if stall > testkeys.Simple.MinPlayoutDelay {
+			testkeys.Simple.MinPlayoutDelay = stall
+		}
+		testkeys.ReceiverData = append(testkeys.ReceiverData, results)
+	}
+
+	median, err := stats.Median(rates)
+	testkeys.Simple.MedianBitrate = int64(median)
+	if err != nil {
+		testkeys.Failure = err.Error()
+		return err
+	}
+	if client.Error() != nil {
+		testkeys.Failure = err.Error()
+		return client.Error()
+	}
+	data, err := json.Marshal(client.ServerResults())
+	if err != nil {
+		testkeys.Failure = err.Error()
+		return err
+	}
+	sess.Logger.Debugf("Server result: %s", string(data))
+	callbacks.OnProgress(1, "done")
+	testkeys.ClientResolver = sess.Location.ResolverIP
```

Comment:
This is not needed anymore, we're now using a toplevel key for that

Reply:
In the midst of the clashes, they did not change again. :(

Reply:
```suggestion
	testkeys.ClientResolver = sess.ResolverIP()
```
?
Or delete?
I don't understand: ` we're now using a toplevel key for that`

Reply:
I meant: delete the line. The code that calls the test is now doing that. Not in the test keys but in the parent object (Measurement). For this reason I said "top level keys", which I can understand why it was not clear at all :)